lib/services/access_request.go

```diff
@@ -1668,51 +1668,52 @@ func roleAllowsResource(
 type ListResourcesRequestOption func(*proto.ListResourcesRequest)
 
 func GetResourceDetails(ctx context.Context, clusterName string, lister ResourceLister, ids []types.ResourceID) (map[string]types.ResourceDetails, error) {
-	var nodeIDs []types.ResourceID
+	var resourceIDs []types.ResourceID
 	for _, resourceID := range ids {
-		if resourceID.Kind != types.KindNode {
-			// The only detail we want, for now, is the server hostname, so we
-			// can skip all other resource kinds as a minor optimization.
-			continue
+		// We're interested in hostname or friendly name details. These apply to
+		// nodes, app servers, and user groups.
+		switch resourceID.Kind {
+		case types.KindNode, types.KindApp, types.KindUserGroup:
+			resourceIDs = append(resourceIDs, resourceID)
 		}
-		nodeIDs = append(nodeIDs, resourceID)
 	}
 
 	withExtraRoles := func(req *proto.ListResourcesRequest) {
 		req.UseSearchAsRoles = true
 		req.UsePreviewAsRoles = true
 	}
 
-	resources, err := GetResourcesByResourceIDs(ctx, lister, nodeIDs, withExtraRoles)
+	resources, err := GetResourcesByResourceIDs(ctx, lister, resourceIDs, withExtraRoles)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	result := make(map[string]types.ResourceDetails)
 	for _, resource := range resources {
-		hn, ok := resource.(interface{ GetHostname() string })
-		if !ok {
+		friendlyName := FriendlyName(resource)
+
+		// No friendly name was found, so skip to the next resource.
+		if friendlyName == "" {
 			continue
 		}
+
 		id := types.ResourceID{
 			ClusterName: clusterName,
 			Kind:        resource.GetKind(),
 			Name:        resource.GetName(),
 		}
 		result[types.ResourceIDToString(id)] = types.ResourceDetails{
-			Hostname: hn.GetHostname(),
+			FriendlyName: friendlyName,
```

Comment:
Yep, I think you're right. I guess we'll have to populate both?

Reply:
i took a look at it again, this info is not sent on the wire, so i think this is fine. but i did saw this: https://github.com/gravitational/teleport/blob/553eb02bdfd73bbd5f20b4871a1e32a7c41ccc42/proto/teleport/lib/teleterm/v1/access_request.proto#L65,

@avatus do we need friendly name defined for teleterm in this proto file?

otherwise ignore my comments (i missed that there were no proto changes)

Reply:
We do need it defined in the proto and some more work to get displayed, but without it, the column will just be hidden. 

Although, let's do the Connect portion in a separate PR I think. I can get a PR up later today or help @mdwn set it up. 

We can leave this as is here and fast follow with the above PR.

Reply:
PR'd to this PR with the updates to Connect. I for some reason thought it would be more complicated, idk why I remembered it being rough for grpc. (I think i was just new when I worked on this originally lol)
https://github.com/gravitational/teleport/pull/27694

This will bring the feature to Connect
![Screenshot 2023-06-09 at 1 45 20 PM](https://github.com/gravitational/teleport/assets/5201977/f1aeda27-175f-449c-b7dd-4a27b4728e5b)